espnet2/svs/singing_tacotron/decoder.py

```diff
@@ -0,0 +1,450 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 Nagoya University (Tomoki Hayashi)
+# Copyright 2023 Renmin University of China (Yuning Wu)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""Singing Tacotron decoder related modules."""
+
+import six
+import torch
+import torch.nn.functional as F
+
+from espnet.nets.pytorch_backend.rnn.attentions import AttForwardTA
+from espnet.nets.pytorch_backend.tacotron2.decoder import Postnet, Prenet, ZoneOutCell
+
+
+def decoder_init(m):
+    """Initialize decoder parameters."""
+    if isinstance(m, torch.nn.Conv1d):
+        torch.nn.init.xavier_uniform_(m.weight, torch.nn.init.calculate_gain("tanh"))
+
+
+class Decoder(torch.nn.Module):
```

Comment:
Since this is very similar to tacotron, do you think if we are possible to inherit the tactotron decoder to minimize the code usage?

Reply:
For decoder, the transition token in GDCA makes it different from tacotron's.